File: ignite-nest/src/post/infrastructure/web/dto/response/post-detail.dto.ts
import { PickType } from '@nestjs/swagger';
import { Post } from '../../../../entity/post.entity';
import { CommentListDto } from '../../../../../comment/infrastructure/web/dto/response/comment-list.dto';

export class PostDetailDto extends PickType(Post, [
  'id',
  'title',
  'content',
  'createdAt',
]) {
  comments: CommentListDto[];
}

File: ignite-spring/src/main/kotlin/org/rookedsysc/ignitespring/post/infrastructure/web/dto/PostDetailDto.kt
package org.rookedsysc.ignitespring.post.infrastructure.web.dto

import org.rookedsysc.ignitespring.comment.infrastructure.web.dto.CommentPostDetailListDto
import java.util.Date

data class PostDetailDto(
    val id: Long,
    val title: String,
    val content: String,
    val createdAt: Date,
    val postDetailListDto: List<CommentPostDetailListDto>
) {}
